diff --git a/var/spack/repos/builtin/packages/gnupg/package.py b/var/spack/repos/builtin/packages/gnupg/package.py
index bcf0549472..d284551143 100644
--- a/var/spack/repos/builtin/packages/gnupg/package.py
+++ b/var/spack/repos/builtin/packages/gnupg/package.py
@@ -125,6 +125,9 @@ class Gnupg(AutotoolsPackage):
     # Getting some linking error.
     conflicts("%gcc@10:", when="@:1")
 
+    def setup_build_environment(self, env):
+        env.set("LDFLAGS", "-lrt")
+
     @run_after("install")
     def add_gpg2_symlink(self):
         if self.spec.satisfies("@2.0:2"):
@@ -132,6 +135,7 @@ def add_gpg2_symlink(self):
 
     def configure_args(self):
         args = [
+            "ac_cv_func_inotify_init=no",
             "--disable-nls",
             "--disable-bzip2",
             "--disable-ldap",
