diff --git a/var/spack/repos/builtin/packages/gnupg/package.py b/var/spack/repos/builtin/packages/gnupg/package.py
index 6e512a105e..5d4a96beaf 100644
--- a/var/spack/repos/builtin/packages/gnupg/package.py
+++ b/var/spack/repos/builtin/packages/gnupg/package.py
@@ -146,6 +146,7 @@ def configure_args(self):
             "--disable-regex",
             "--with-zlib=" + self.spec["zlib-api"].prefix,
             "--without-tar",
+            "--without-libintl-prefix",
             "--without-readline",
         ]
 
diff --git a/var/spack/repos/builtin/packages/libgpg-error/package.py b/var/spack/repos/builtin/packages/libgpg-error/package.py
index e2eeb228b4..23282a2883 100644
--- a/var/spack/repos/builtin/packages/libgpg-error/package.py
+++ b/var/spack/repos/builtin/packages/libgpg-error/package.py
@@ -36,6 +36,8 @@ def configure_args(self):
         args = [
             "--enable-static",
             "--enable-shared",
+            "--without-libintl-prefix",
+            "--disable-nls",
             "--enable-tests" if self.run_tests else "--disable-tests",
         ]
         if self.spec.satisfies("@1.46:"):
