diff --git a/var/spack/repos/builtin/packages/libgpg-error/package.py b/var/spack/repos/builtin/packages/libgpg-error/package.py
index acdc73f8a7..6deb0253d4 100644
--- a/var/spack/repos/builtin/packages/libgpg-error/package.py
+++ b/var/spack/repos/builtin/packages/libgpg-error/package.py
@@ -32,6 +32,7 @@ class LibgpgError(AutotoolsPackage):
 
     def configure_args(self):
         return [
+            "--disable-nls",
             "--enable-static",
             "--enable-shared",
             "--enable-tests" if self.run_tests else "--disable-tests",