diff --git a/var/spack/repos/builtin/packages/gnupg/package.py b/var/spack/repos/builtin/packages/gnupg/package.py
index 9b1a4c64db..4da9a12328 100644
--- a/var/spack/repos/builtin/packages/gnupg/package.py
+++ b/var/spack/repos/builtin/packages/gnupg/package.py
@@ -77,6 +77,7 @@ def configure_args(self):
             "--disable-regex",
             f"--with-zlib={self.spec['zlib-api'].prefix}",
             "--without-tar",
+            "--without-libintl-prefix",
             "--without-readline",
         ]
 
diff --git a/var/spack/repos/builtin/packages/libgpg-error/package.py b/var/spack/repos/builtin/packages/libgpg-error/package.py
index b28ada157b..22c48325cf 100644
--- a/var/spack/repos/builtin/packages/libgpg-error/package.py
+++ b/var/spack/repos/builtin/packages/libgpg-error/package.py
@@ -39,6 +39,8 @@ def configure_args(self):
         args = [
             "--enable-static",
             "--enable-shared",
+            "--without-libintl-prefix",
+            "--disable-nls",
             "--enable-tests" if self.run_tests else "--disable-tests",
         ]
         if self.spec.satisfies("@1.46:"):
